--- opendkim.py--ORG	2019-04-25 15:35:19.000000000 +0200
+++ opendkim.py	2019-04-25 15:34:37.000000000 +0200
@@ -74,8 +74,8 @@
             dbname, dbuser, dbpassword,
             self.get_file_path("dkim_view_{}.sql".format(self.dbengine))
         )
-        self.backend.grant_right_on_table(
-            dbname, "dkim", self.app_config["dbuser"], "SELECT")
+        #self.backend.grant_right_on_table(
+        #    dbname, "dkim", self.app_config["dbuser"], "SELECT")
 
     def post_run(self):
         """Additional tasks."""
