--- package.py--ORG	2019-04-14 09:50:49.000000000 +0200
+++ package.py	2019-04-14 09:50:57.000000000 +0200
@@ -30,7 +30,7 @@
         """Update local cache."""
         if self.index_updated:
             return
-        utils.exec_cmd("apt-get update --quiet")
+        #utils.exec_cmd("apt-get update --quiet")
         self.index_updated = True
 
     def preconfigure(self, name, question, qtype, answer):
