--- modoboa_radicale/backends/caldav_.py--ORG	2019-04-20 18:13:05.226010952 +0200
+++ modoboa_radicale/backends/caldav_.py	2019-04-20 18:07:39.080445942 +0200
@@ -25,7 +25,7 @@
             param_tools.get_global_parameter("server_location"))
         self.client = caldav.DAVClient(
             server_url,
-            username=username, password=password)
+            username=username, password=password, ssl_verify_cert=True)
         if self.calendar:
             self.remote_cal = self.client.calendar(calendar.encoded_path)
 
